### ISSUE TITLE
Improved Settings Collection UI

### ISSUE THREAD

**Issue by konzz** [huridocs/uwazi#3480]
17 Points AVG

- [x] Apply the designs for the new collection settings.
- [x] Check Mobile
- [x] Migrate the form  library (react-redux-forms) to react-hook-form
- [x] Split the form in smaller components

https://www.figma.com/file/xjsz7r65epN4B0b96zilac/0-uwazi-redesigns?node-id=1554%3A1

**Comment by nwyu** [comment]
Related [#2938](https://github.com/huridocs/uwazi/issues/2938)

**Comment by RafaPolit** [comment]
We have finished the development, and also updated the e2e tests that covered this area, both on the nightmare and puppeteer folders.  Puppeteer tests are working nicely.

Unfortunately, we have stumbled upon an impassable hurdle: react-hook-form uses internally a `.map(...).flat` implementation of arrays that is not supported by our current versions of electron / nightmare / jest, not sure which is the culprit.

We have tried to update nightmare.  V2.10 is our current version and the last 2.x version developed.  Upgrading to 3.x has proved not at all straightforward.  Same as with upgrading jest.

There are two scenarios that may help finish this task:
- migrate current tests pertaining to collection settings into puppeteer (probably a good thing altogether)
- invest more time trying to update nightmare, which is kind of a nightmare :) and we have already deprecated that flow for e2e

That is our current status.  It is a shame, as the functionality is all finished.
